framework_tests/test_couchbase_server_install.py

```diff
@@ -0,0 +1,35 @@
+from libraries.provision.install_couchbase_server import CouchbaseServerConfig
+
+
+def test_version_only():
+    server_config = CouchbaseServerConfig(
+        version="4.1.1"
+    )
+    assert server_config.version == "4.1.1"
+    assert server_config.build is None
+
+
+def test_version_build():
+    server_config = CouchbaseServerConfig(
+        version="4.1.1-5914"
+    )
+    assert server_config.version == "4.1.1"
+    assert server_config.build == "5914"
+
+
+def test_enterprise():
+    server_config = CouchbaseServerConfig(
+        version="4.1.1-5914"
+    )
+    base_url, package_name = server_config.get_baseurl_package()
+    assert base_url == "http://cbnas01.sc.couchbase.com/builds/latestbuilds/couchbase-server/sherlock/5914"
```

Comment:
this feels like it has a dependency on a hardcoded value.  

what if it just asserted that the base_url had 5914 somewhere in it?  likewise, assert that `package_name` had 4.1.1 somewhere in the string?